src/main/java/us/kbase/auth/client/AuthClient.java

```diff
@@ -179,4 +186,53 @@ public AuthToken validateToken(final String token) throws IOException, AuthExcep
 		return authToken;
 	}
 
+	private void checkToken(final String token) {
+		if (token == null || token.trim().isEmpty()) {
+			throw new IllegalArgumentException("token must be a non-whitespace string");
+		}
+	}
+	
+	/** Check if usernames are valid accounts in the auth service.
+	 * @param users the list of usernames to check. If they contain any invalid characters
+	 * (e.g. anything other than a-z, 0-9, or _, an exception will be thrown.
+	 * @param token any valid auth token.
+	 * @return a mapping of each username to whether it's valid or not.
+	 * @throws IOException if an IOException occurs communicating with the auth service.
+	 * @throws AuthException if an auth exception occurs communicating with the auth service.
+	 */
+	public Map<String, Boolean> isValidUserName(final List<String> users, final String token)
+			throws IOException, AuthException {
+		// theoretically someone could submit hundreds of users and hit the url size limit
+		// don't worry about that for now.
+		checkToken(token);
```

Comment:
Do we need a valid token for this? I can see a case (mainly at account creation) where you don't have your own username yet, thus no token (maybe a temp token via an auth provider?), and you want to see if your username is available or not.

Reply:
Yes, this endpoint requires a token. For the use case you're talking about there's https://github.com/kbase/auth2/blob/develop/src/main/java/us/kbase/auth2/service/ui/Login.java#L136-L150